analyzer/runtime/extract_test.go

```diff
@@ -0,0 +1,48 @@
+package runtime
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestPlaintextError(t *testing.T) {
+	inputs := []string{
+		"error",
+		"execution failed: out of fund",
+		`reverted: invalid reason prefix: 'EAlgywAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAM'`,
+	}
+	for _, input := range inputs {
+		msg := tryParseErrorMessage(input)
+		require.NotNil(t, msg, "failed to extract error message")
+		require.Equal(t, input, *msg, "parsed message should match original input message")
+	}
+}
+
+func TestKnownStringError(t *testing.T) {
+	inputs := map[string]string{
+		"too late to submit": "reverted: CMN5oAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABJ0b28gbGF0ZSB0byBzdWJtaXQAAAAAAAAAAAAAAAAAAA==",
```

Comment:
Because the tx can fail for reasons other than being reverted, we should keep the "reverted: " prefix when decoding the base64 stuff. In other words, the correct decoding here would be "reverted: too late to submit".

Reply:
got it, updated the fn to include the `reverted: `